google/resource_dataproc_cluster_test.go

```diff
@@ -0,0 +1,780 @@
+
+	    # Additional ones specifically desired by user (Note for now must be in alpha order
+	    # of fully qualified scope name)
+	    "https://www.googleapis.com/auth/monitoring"
+
+	]
+
+}`, rnd, saEmail)
+}
+
+func testAccDataprocCluster_withNetworkRefs(rnd, netName string) string {
+	return fmt.Sprintf(`
+resource "google_compute_network" "dataproc_network" {
+	name = "%s"
+	auto_create_subnetworks = true
+}
+
+resource "google_compute_firewall" "dataproc_network_firewall" {
+	name = "dproc-cluster-test-%s-allow-internal"
+	description = "Firewall rules for dataproc Terraform acceptance testing"
+	network = "${google_compute_network.dataproc_network.name}"
+
+	allow {
+	    protocol = "icmp"
+	}
+
+	allow {
+		protocol = "tcp"
+		ports    = ["0-65535"]
+	}
+
+	allow {
+		protocol = "udp"
+		ports    = ["0-65535"]
+	}
+}
+
+resource "google_dataproc_cluster" "with_net_ref_by_name" {
+	name   = "dproc-cluster-test-%s-name"
+	region = "us-central1"
+	depends_on = ["google_compute_firewall.dataproc_network_firewall"]
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+
+	network = "${google_compute_network.dataproc_network.name}"
+}
+
+resource "google_dataproc_cluster" "with_net_ref_by_url" {
+	name   = "dproc-cluster-test-%s-url"
+	region = "us-central1"
+    depends_on = ["google_compute_firewall.dataproc_network_firewall"]
+
+    # Keep the costs down with smallest config we can get away with
+    properties = {
+        "dataproc:dataproc.allow.zero.workers" = "true"
+    }
+    worker_config { }
+	master_config {
+		machine_type      = "n1-standard-1"
+		boot_disk_size_gb = 10
+	}
+
+	network = "${google_compute_network.dataproc_network.self_link}"
+}
+
+`, netName, rnd, rnd, rnd)
+}
+
+func testAccPreCheckWithServiceAccount(t *testing.T) {
+	testAccPreCheck(t)
+	if v := os.Getenv("GOOGLE_SERVICE_ACCOUNT"); v == "" {
+		t.Skipf("GOOGLE_SERVICE_ACCOUNT must be set for the dataproc acceptance test testing service account functionality")
+	}
+
+}
+
+func validateDataprocCluster_withConfigOverrides(n string, cluster *dataproc.Cluster) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		type tfAndGCPTestField struct {
+			tfAttr       string
+			expectedVal  string
+			actualGCPVal string
+		}
+
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Terraform resource Not found: %s", n)
+		}
+
+		if cluster.Config.MasterConfig == nil || cluster.Config.WorkerConfig == nil || cluster.Config.SecondaryWorkerConfig == nil {
+			return fmt.Errorf("Master/Worker/SecondaryConfig values not set in GCP, expecting values")
+		}
+
+		clusterTests := []tfAndGCPTestField{
+			{"master_config.0.num_masters", "3", strconv.FormatInt(cluster.Config.MasterConfig.NumInstances, base10)},
```

Comment:
Can't use `strconv.Itoa` directly as its an int64 however as the type of fields being used are really not going to get that big, in reality I can probably downcast without loss of precision